matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/DefaultCryptoService.kt

```diff
@@ -670,61 +682,73 @@ internal class DefaultCryptoService @Inject constructor(
+                                    ?.let {
+                                        markOlmSessionForUnwedging(event.senderId ?: "", it)
+                                    }
+                                    ?: run {
+                                        Timber.v("## CRYPTO | markOlmSessionForUnwedging() : Failed to find sender crypto device")
+                                    }
+                        }
                     }
                     throw mxCryptoError
                 }
             }
         }
     }
 
-    private fun markOlmSessionForUnwedging(event: Event, mxMegolmDecryption: MXMegolmDecryption) {
-        val senderId = event.senderId ?: return
-        val encryptedMessage = event.content.toModel<EncryptedEventContent>() ?: return
-        val deviceKey = encryptedMessage.senderKey ?: return
-        encryptedMessage.algorithm?.takeIf { it == MXCRYPTO_ALGORITHM_MEGOLM } ?: return
-
-        if (senderId == userId
-                && deviceKey == olmDevice.deviceCurve25519Key) {
-            Timber.d("[MXCrypto] markOlmSessionForUnwedging: Do not unwedge ourselves")
-            return
-        }
-
-        val lastForcedDate = lastNewSessionForcedDates.getObject(senderId, deviceKey) ?: 0
-        val now = System.currentTimeMillis()
-        if (now - lastForcedDate < CRYPTO_MIN_FORCE_SESSION_PERIOD_MILLIS) {
-            Timber.d("[MXCrypto] markOlmSessionForUnwedging: New session already forced with device at $lastForcedDate. Not forcing another")
-            return
-        }
-
-        // Establish a new olm session with this device since we're failing to decrypt messages
-        // on a current session.
-        val deviceInfo = getDeviceInfo(senderId, deviceKey) ?: return Unit.also {
-            Timber.d("[MXCrypto] markOlmSessionForUnwedging: Couldn't find device for identity key $deviceKey: not re-establishing session")
-        }
-
-        Timber.d("[MXCrypto] markOlmSessionForUnwedging from $senderId:${deviceInfo.deviceId}")
-        lastNewSessionForcedDates.setObject(senderId, deviceKey, now)
-
-        mxMegolmDecryption.markOlmSessionForUnwedging(senderId, deviceInfo)
-    }
+//    private fun markOlmSessionForUnwedging(event: Event, mxMegolmDecryption: MXMegolmDecryption) {
+//        Timber.d("## CRYPTO |  markOlmSessionForUnwedging: ${event.eventId}")
+//        val senderId = event.senderId ?: return
+//        val encryptedMessage = event.content.toModel<EncryptedEventContent>() ?: return
+//        val deviceKey = encryptedMessage.senderKey ?: return
+//        encryptedMessage.algorithm?.takeIf { it == MXCRYPTO_ALGORITHM_MEGOLM } ?: return
+//
+//        if (senderId == userId
+//                && deviceKey == olmDevice.deviceCurve25519Key) {
+//            Timber.d("## CRYPTO |  markOlmSessionForUnwedging: Do not unwedge ourselves")
+//            return
+//        }
+//
+//        val lastForcedDate = lastNewSessionForcedDates.getObject(senderId, deviceKey) ?: 0
```

Comment:
the date management is not used anymore then?

Reply:
good catch, didn't know it was implemented